libs/networking/http_helpers/http_helpers.ml

```diff
@@ -223,15 +170,66 @@ struct
        always respond with TLS v1.3, as we are not guaranteed to hit the same
        node. Currently, AWS does not support specifying a minimum TLS version
        of v1.3, and we will need to figure out a better solution for ensuring
-       reliable metrics delivery.
-    *)
-    Lwt_platform.run
-      (Lwt.catch
-         (fun () -> post_async ~body ~headers ~chunked caps url)
-         (fun exn ->
-           let err = Printexc.to_string exn in
-           Log.err (fun m ->
-               m "send to '%s' failed: %s" (Uri.to_string url) err);
-           Lwt.return (Error (0, err))))
-  [@@profiling]
-end
+       reliable metrics delivery. *)
+    try
+      let%lwt responses_stream = Client.callv url stream_req in
+      Lwt.return_ok responses_stream
+    with
+    | exn ->
+        let err = Printexc.to_string exn in
+        Log.err (fun m ->
+            m "HTTP %S to '%s' failed: %s" (string_of_meth meth)
+              (Uri.to_string url) err);
+        Lwt.return_error err
+  in
+  let handle_responses responses_stream =
+    (* Assume that we only get one response back *)
+    (* MUST USE GET HERE *)
+    let%lwt repsonses =
+      responses_stream |> Lwt_stream.map_s resp_handler |> Lwt_stream.to_list
+    in
+    match repsonses with
+    | [ (response, response_body) ] -> Lwt.return_ok (response, response_body)
+    | [] -> Lwt.return_error "No responses from a single request"
+    | _ -> Lwt.return_error "Multiple responses from a single request"
+  in
+  Lwt_result.bind responses_stream_opt handle_responses
+
+(*****************************************************************************)
+(* Async *)
+(*****************************************************************************)
+let rec get ?(headers = []) caps url =
+  Log.info (fun m -> m "GET on %s" (Uri.to_string url));
+  (* This checks to make sure a client has been set *)
+  (* Instead of defaulting to a client, as that can cause *)
+  (* Hard to debug build and runtime issues *)
+  let response_result = call_client ~headers `GET url in
+  let handle_response (response, body) =
+    let server_response = server_response_of_response (response, body) `GET in
+    match server_response.code with
+    | 307 -> (
+        let location = Header.get (response |> Response.headers) "location" in
+        match location with
+        | None ->
+            let code_str = Code.string_of_status response.status in
+            let err = "HTTP GET failed: " ^ code_str ^ ":\n" ^ body in
+            Log.err (fun m -> m "%s" err);
+            let server_response = { server_response with body = Error err } in
+            Lwt.return_ok server_response
+        | Some url -> get caps (Uri.of_string url))
+    | _ -> Lwt.return_ok server_response
+  in
+  Lwt_result.bind response_result handle_response
+[@@profiling]
+
+let post ~body ?(headers = [ ("content-type", "application/json") ])
```

Comment:
Not a huge fan of this as a default header. Fine if we want to leave it since this was in the old version too though.

Reply:
let's just leave it, I think that it's net positive for us